parlai/tasks/blended_skill_talk/worlds.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import random
+import os
+from copy import deepcopy
+
+from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.tasks.self_chat.worlds import SelfChatWorld as SelfChatBaseWorld
+
+
+def load_personas(opt):
+    print('[ loading personas.. ]')
+    fname = os.path.join(
+        opt['datapath'], 'blended_skill_talk', opt['datatype'] + '.json'
+    )
+    with open(fname) as json_file:
+        data = json.load(json_file)
+    contexts = []
+    for d in data:
+        p1 = (
+            'your persona: '
+            + d['personas'][0][0]
+            + '\n'
+            + 'your persona: '
+            + d['personas'][0][1]
+            + '\n'
+        )
+        p2 = (
+            'your persona: '
+            + d['personas'][1][0]
+            + '\n'
+            + 'your persona: '
+            + d['personas'][1][1]
+            + '\n'
+        )
+        if d['context_dataset'] == 'wizard_of_wikipedia':
+            p1 += d['additional_context'] + '\n'
+            p2 += d['additional_context'] + '\n'
+        ctxt = d['free_turker_utterance'] + '\n' + d['guided_turker_utterance']
+        p1 += ctxt
+        p2 += ctxt
+        contexts.append([p1, p2])
+    return contexts
+
+
+class InteractiveWorld(DialogPartnerWorld):
+    @staticmethod
+    def add_cmdline_args(argparser):
+        parser = argparser.add_argument_group('BST Interactive World')
+        parser.add_argument(
+            '--display-partner-persona',
+            type='bool',
+            default=True,
+            help='Display your partner persona at the end of the chat',
+        )
+
+    def __init__(self, opt, agents, shared=None):
+        super().__init__(opt, agents, shared)
+        self.contexts = load_personas(self.opt)
+        self.display_partner_persona = self.opt['display_partner_persona']
+        self.cnt = 0
+
+    def get_new_personas(self):
+        random.seed()
+        p = random.choice(self.contexts)
+        return p[0], p[1]
+
+    def parley(self):
```

Comment:
i don't think you need any of this given @jaseweston 's recent changes. see #2497

Reply:
Yes - subclassed his `InteractiveWorld`